### PR TITLE
support passing a config file as a CLI argument

### PR DESCRIPTION
**Pull request by teto**
- **add flake**
- **Load config-file from an alternate path**
Attempt at addressing https://github.com/fourmolu/fourmolu/issues/396

### CODE REVIEW

File: app/Main.hs [hunk 1]
```diff
@@ -272,6 +274,8 @@ data Opts = Opts
     optQuiet :: !Bool,
     -- | Ormolu 'Config'
     optConfig :: !(Config RegionIndices),
+    -- | Ormolu 'Config'
```
Comment (brandonchinn178) [inline app/Main.hs:277]:
```suggestion
    -- | Path to Fourmolu config file
```

File: app/Main.hs [hunk 2]
```diff
@@ -361,6 +365,11 @@ optsParser =
         help "Make output quieter"
       ]
     <*> configParser
+    <*> (optional . strOption . mconcat)
+      [ metavar "CONFIG_FILE",
+        long "config-file",
```
Comment (brandonchinn178) [inline app/Main.hs:370]: Any reason why not just `--config`? I think that's more standard with other CLI tools

File: app/Main.hs [hunk 3]
```diff
@@ -43,10 +43,12 @@ main = do
   opts@Opts {..} <- execParser optsParserInfo
 
   cwd <- getCurrentDirectory
-  cfg <- case optInputFiles of
-    [] -> mkConfig cwd opts
-    ["-"] -> mkConfig cwd opts
-    file : _ -> mkConfig file opts
+  cfg <- case optConfigFilePath of 
+    Just configFilePath -> mkConfig configFilePath opts
```
Comment (brandonchinn178) [inline app/Main.hs:47]: I don't think this works? Did you test this?

I'm assuming you want to be able to do `--config-file /other/path/to/other_config_name_fourmolu.yaml`, but `mkConfig` will still look for a file named `fourmolu.yaml` in the directory named `configFilePath`.

I think this PR requires two changes:
1. Break up `loadConfigFile` into `findConfigFile` that does the logic to search for `fourmolu.yaml` and `loadConfigFile`, which should just decode the config file
2. Change `mkConfig` to take in `ConfigFileLoadResult` and in `main`, get that result from either `--config-file` => `loadConfigFile`, or `findConfigFile (cwd || head optInputFiles)`

File: flake.nix [hunk 4]
```diff
@@ -0,0 +1,84 @@
+{
```
Comment (brandonchinn178) [inline flake.nix:1]: Is there a particular reason why this is being added? None of the current Fourmolu maintainers use Nix, and this file isn't being used in CI, so it'll be very prone to bitrot.

Comment (teto) [reply flake.nix:1]: sry I forgot to remove it.

File: app/Main.hs (hunk 3 above)
Comment (teto) [reply app/Main.hs]: done

File: src/Ormolu/Config.hs [hunk 5]
```diff
@@ -267,19 +268,22 @@ emptyConfig =
...
   findFile dirs configFileName >>= \case
     Nothing -> return $ ConfigNotFound dirs
-    Just file ->
-      either (ConfigParseError file) (ConfigLoaded file)
-        <$> Yaml.decodeFileEither file
+    Just file -> loadConfigFile file
+
+-- | Read options from a config file, if found.
+-- Looks recursively in parent folders, then in 'XdgConfig',
+-- for a file named /fourmolu.yaml/.
```
Comment (brandonchinn178) [inline src/Ormolu/Config.hs:282]: I believe this "look recursively" part should be documented for `findConfigFile` instead

**Comment by brandonchinn178** [comment]
Reimplementing it here: https://github.com/fourmolu/fourmolu/pull/415

**Pull request closed without merging**